pgbot/commands/admin/__init__.py

```diff
@@ -1335,6 +1335,175 @@ async def cmd_react(self, message: discord.PartialMessage, *emojis: str):
         except discord.NotFound:
             pass
 
+    @no_dm
+    async def cmd_browse(
+        self,
+        channel: discord.TextChannel,
+        quantity: Optional[int] = None,
+        before: Optional[Union[discord.Message, datetime.datetime]] = None,
+        after: Optional[Union[discord.Message, datetime.datetime]] = None,
+        around: Optional[Union[discord.Message, datetime.datetime]] = None,
+        controllers: Optional[tuple[discord.Member, ...]] = None,
+    ):
+        """
+        ->type More admin commands
+        ->signature pg!browse <channel> <quantity> [before=] [after=] [around=]
+        ->description Browse through Discord messages
+
+        ->extended description
+        Function to help browse through discord messages
+
+        __Args__:
+            `channel: discord.TextChannel`
+            > A Discord message that reactions should be added to.
+
+            `quantity: int`
+            > The number of messages to get
+
+            `before: Time | Message`
+            > The message or time before which the messages should be shown.
+
+            `after: Time | Message`
+            > The message or time after which the messages should be shown.
+
+            `around: Time | Message`
+            > The message or time around which the messages should be shown.
+
+            `controllers: User | tuple[user]`
+            > The user(s) who can control the embed.
+
+        __Raises__:
+            > `BotException`: One or more given arguments are invalid.
+            > `HTTPException`: An invalid operation was blocked by Discord.
+        -----
+        """
+        # needed for typecheckers to know that self.author is a member
+        if isinstance(self.author, discord.User):
+            return
+
+        max_messages = 500
```

Comment:
This should prolly be a constant (yes, please scream MAX_MESSAGES, and scream it in common.py where other constants are screaming and not here) 😉